docs/monadtransformers/index.md

```diff
@@ -0,0 +1,213 @@
+
+`EitherT` is very useful for fail-fast chained verifications:
+
+```scala mdoc:silent
+
+case class Item(state: String)
+class ItemOrder  { /* ... */ }
+
+case class ItemNotFoundException(message: String) extends BaseException(message)
+case class InvalidItemStateException(message: String) extends BaseException(message)
+
+def getItemById(itemId: Int): Try[Option[Item]] = { /* .. */ ??? }
+
+def ensureItemExists(itemId: Int): EitherT[Try, BaseException, Item] = {
+  OptionT(getItemById(itemId))
+    .toRight(ItemNotFoundException(s"item not found, itemId = $itemId"))
+}
+
+def ensureItemStateIs(actual: String, expected: String): EitherT[Try, BaseException, Unit] = {
+  // Returns a Unit value wrapped into Right and then into Try if condition is true,
+  // otherwise the provided exception wrapped into Left and then into Try.
+  EitherT.cond(actual == expected, (), InvalidItemStateException(s"actual=$actual, expected=$expected"))
+}
+
+def placeOrderForItem(userId: Int, itemId: Int, count: Int): Try[ItemOrder] = { /* ... */ ??? }
+
+def buyItem(userId: Int, itemId: Int, count: Int): EitherT[Try, BaseException, ItemOrder] = {
+  for {
+    user <- ensureUserExists(userId)
+    item <- ensureItemExists(itemId)
+    _ <- ensureItemStateIs(item.state, "AVAILABLE_IN_STOCK")
+    // EitherT.liftF is necessary to make EitherT[Try, BaseException, ItemOrder] out of Try[ItemOrder]
+    placedOrder <- EitherT.liftF(placeOrderForItem(userId, itemId, count))
+  } yield placedOrder
+}
+```
+
+In the above example, we're running various checks against the item one by one.
+If any of the checks fails, the resulting `EitherT` will contain a `Left` value.
+Otherwise, if all of the checks yield a `Right` (of course we mean a `Right` wrapped into an `EitherT`), then the final outcome will also contain `Right`.
+This is a fail-fast behavior: we're effectively stopping the `for` comprehension flow at the first `Left`-ish result.
+
+If you're instead looking for validation that accumulates the errors (e.g. when dealing with user-provided form data), [Validated](../datatypes/validated.md) or the [Parallel](../typeclasses/parallel.md) applicative functor for `Either` may be a good choice.
+
+## Pitfalls
+
+In principle, monad transformers can be stacked to create data types with complex behavior.
+While powerful, this can lead to unwieldy type signatures and poor type inference. 
+The [Cats MTL](https://typelevel.org/cats-mtl/) project provides a capability based approach to extend effect types without requiring as much boilerplate.
+
+Some monad transformers can also interact poorly with more powerful effect types that provide concurrent computation, such as Cats Effect's `IO` or fs2's `Stream`. You should avoid `StateT` and `WriterT` and use the concurrent data types provided by those libraries instead. Because `IO` already provides its own error channel, `EitherT[IO, Throwable]` can also lead to confusing behavior; prefer `IO`'s own error channel instead.
```

Comment:
I think this warning should definitely be repeated on the `EitherT`, `StateT`, and `WriterT`, pages.

Reply:
Done

![image](https://github.com/user-attachments/assets/b8863204-5ba8-404d-897e-e6e35d63ac7c)
![image](https://github.com/user-attachments/assets/8b27b9c3-5101-47d7-8e40-590cafd2fe4e)
![image](https://github.com/user-attachments/assets/924b078d-0e9d-4218-b8d6-61c64e646a49)

Reply:
Thanks for the screenshots.
I think these are good.
The warning on `EitherT` is a little too up top front and center, but I don't think that's a blocker.

Reply:
I couldn't identify a nice place to put that one, EitherT doesn't have a convenient intro like the other ones.